src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5766,13 +5786,15 @@ class Parser {
         }
 
         const bool previousAllowNewTarget = context->allowNewTarget;
+        const bool previousHasParameterOtherThanIdentifier = this->currentScopeContext->m_hasParameterOtherThanIdentifier;
 
         context->allowNewTarget = true;
         MetaNode node = this->createNode();
         StatementContainer* params = this->parseFunctionParameters(builder);
         BlockStatementNode* body = this->parseFunctionBody(builder);
 
         context->allowNewTarget = previousAllowNewTarget;
+        this->currentScopeContext->m_hasParameterOtherThanIdentifier = previousHasParameterOtherThanIdentifier;
```

Comment:
These two lines are unnecessay. We check strict directive during the scanning of the whole souce code but `parseScriptFunction` is called when a function is actually called.
Other looks great :)